test/test_pghoard.py

```diff
@@ -755,6 +760,61 @@ def test_startup_walk_for_missed_compressed_file_type(self, file_type: FileType,
         upload_event = self.pghoard.transfer_queue.get(timeout=1.0)
         assert upload_event.file_type == file_type
 
+    @patch("pghoard.compressor.wal.verify_wal", Mock())
+    @patch.object(PGReceiveXLog, "run", Mock())
+    @patch.object(TransferAgent, "get_object_storage")
+    def test_graceful_shutdown_with_partial_wal_file(
+        self,
+        mocked_get_object_storage: MagicMock,
+    ) -> None:
+        compressed_wal_path, _ = self.pghoard.create_backup_site_paths(self.test_site)
+        uncompressed_wal_path = compressed_wal_path + "_incoming"
+
+        self.config["backup_sites"][self.test_site]["active_backup_mode"] = "pg_receivexlog"
```

Comment:
Shouldn't we also test with other archival methods ?

Reply:
mmm not really, since I am not actually activating those threads. Actually I should remove this line and the call for `receivexlog_listener` , also rename this test... my initial intention was testing if partial files got renamed, but discarded that idea after running tests on prune (its too dangerous).